hugr-model/capnp/hugr-v0.capnp

```diff
@@ -0,0 +1,175 @@
+@0xe02b32c528509601;
+
+struct Module {
+    root @0 :UInt32;
+    nodes @1 :List(Node);
+    regions @2 :List(Region);
+    terms @3 :List(Term);
+}
+
+struct Node {
+    operation @0 :Operation;
+    inputs @1 :List(LinkRef);
+    outputs @2 :List(LinkRef);
+    params @3 :List(UInt32);
+    regions @4 :List(UInt32);
+    meta @5 :List(MetaItem);
+    signature @6 :UInt32;
+}
+
+struct Operation {
+    union {
+        invalid @0 :Void;
```

Comment:
What's the `invalid` for?

Reply:
See the docs on Term::Invalid. It serves two purposes: - being able to represent modules whose nodes have non-contiguous indices without needing to compact
- a placeholder operation for when you have partially constructed a module. This was useful to break a cycle in export after adding the defining node index in LocalRef.